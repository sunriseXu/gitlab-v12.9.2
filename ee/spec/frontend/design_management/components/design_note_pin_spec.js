import { shallowMount } from '@vue/test-utils';
import DesignNotePin from 'ee/design_management/components/design_note_pin.vue';

describe('Design discussions component', () => {
  let wrapper;

  function createComponent(propsData = {}) {
    wrapper = shallowMount(DesignNotePin, {
      propsData: {
        position: {
          left: '10px',
          top: '10px',
        },
        ...propsData,
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot of note without index', () => {
    createComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match the snapshot of note with index', () => {
    createComponent({ label: '1' });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match the snapshot of note when repositioning', () => {
    createComponent({ repositioning: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('pinStyle', () => {
    it('sets cursor to `move` when repositioning = true', () => {
      createComponent({ repositioning: true });
      expect(wrapper.vm.pinStyle.cursor).toBe('move');
    });

    it('does not set cursor when repositioning = false', () => {
      createComponent();
      expect(wrapper.vm.pinStyle.cursor).toBe(undefined);
    });
  });
});
