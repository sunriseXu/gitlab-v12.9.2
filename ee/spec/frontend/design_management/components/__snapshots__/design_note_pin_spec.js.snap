// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design discussions component should match the snapshot of note when repositioning 1`] = `
<button
  aria-label="Comment form position"
  class="position-absolute btn-transparent comment-indicator"
  style="left: 10px; top: 10px; cursor: move;"
  type="button"
>
  <icon-stub
    name="image-comment-dark"
    size="16"
  />
</button>
`;

exports[`Design discussions component should match the snapshot of note with index 1`] = `
<button
  aria-label="Comment '1' position"
  class="position-absolute js-image-badge badge badge-pill"
  style="left: 10px; top: 10px;"
  type="button"
>
  
    1
  
</button>
`;

exports[`Design discussions component should match the snapshot of note without index 1`] = `
<button
  aria-label="Comment form position"
  class="position-absolute btn-transparent comment-indicator"
  style="left: 10px; top: 10px;"
  type="button"
>
  <icon-stub
    name="image-comment-dark"
    size="16"
  />
</button>
`;
